From 29f914a9810d22bca27ab35478a123e92c1e6ffc Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 2 Oct 2025 09:11:14 +0200
Subject: [PATCH] drm/panel: overrides the reset or enable pin configuration

Temporary patch to configure the reset or enable
pin in output mode.

Change-Id: I7ddd0cd76e13020194100b2674497a5eef3e5646
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/495800
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/panel/panel-himax-hx8394.c       | 2 ++
 drivers/gpu/drm/panel/panel-lvds.c               | 2 ++
 drivers/gpu/drm/panel/panel-orisetech-otm8009a.c | 2 ++
 drivers/gpu/drm/panel/panel-raydium-rm68200.c    | 2 ++
 drivers/gpu/drm/panel/panel-simple.c             | 2 ++
 5 files changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-himax-hx8394.c b/drivers/gpu/drm/panel/panel-himax-hx8394.c
index 5e8537039703..636ceb00f2fa 100644
--- a/drivers/gpu/drm/panel/panel-himax-hx8394.c
+++ b/drivers/gpu/drm/panel/panel-himax-hx8394.c
@@ -500,6 +500,8 @@ static int hx8394_probe(struct mipi_dsi_device *dsi)
 		ret = gpiod_get_direction(ctx->reset_gpio);
 		if (ret != 0)
 			gpiod_direction_output(ctx->reset_gpio, 1);
+		else
+			gpiod_direction_output(ctx->reset_gpio, 0);
 	}
 
 	mipi_dsi_set_drvdata(dsi, ctx);
diff --git a/drivers/gpu/drm/panel/panel-lvds.c b/drivers/gpu/drm/panel/panel-lvds.c
index ba1d936d2c96..8b2c6be3e3a1 100644
--- a/drivers/gpu/drm/panel/panel-lvds.c
+++ b/drivers/gpu/drm/panel/panel-lvds.c
@@ -210,6 +210,8 @@ static int panel_lvds_probe(struct platform_device *pdev)
 		ret = gpiod_get_direction(lvds->enable_gpio);
 		if (ret != 0)
 			gpiod_direction_output(lvds->enable_gpio, 0);
+		else
+			gpiod_direction_output(lvds->enable_gpio, 1);
 	}
 
 	lvds->reset_gpio = devm_gpiod_get_optional(lvds->dev, "reset",
diff --git a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
index 157c3e05e17b..8b74b5b94fc3 100644
--- a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
+++ b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
@@ -463,6 +463,8 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 		ret = gpiod_get_direction(ctx->reset_gpio);
 		if (ret != 0)
 			gpiod_direction_output(ctx->reset_gpio, 1);
+		else
+			gpiod_direction_output(ctx->reset_gpio, 0);
 	}
 
 	ctx->supply = devm_regulator_get(dev, "power");
diff --git a/drivers/gpu/drm/panel/panel-raydium-rm68200.c b/drivers/gpu/drm/panel/panel-raydium-rm68200.c
index 1f7b867bc1ef..cd648cfaaab0 100644
--- a/drivers/gpu/drm/panel/panel-raydium-rm68200.c
+++ b/drivers/gpu/drm/panel/panel-raydium-rm68200.c
@@ -406,6 +406,8 @@ static int rm68200_probe(struct mipi_dsi_device *dsi)
 		ret = gpiod_get_direction(ctx->reset_gpio);
 		if (ret != 0)
 			gpiod_direction_output(ctx->reset_gpio, 1);
+		else
+			gpiod_direction_output(ctx->reset_gpio, 0);
 	}
 
 	ctx->supply = devm_regulator_get(dev, "power");
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index b7697f5a247f..a708bd1566a3 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -603,6 +603,8 @@ static int panel_simple_probe(struct device *dev, const struct panel_desc *desc)
 		err = gpiod_get_direction(panel->enable_gpio);
 		if (err != 0)
 			gpiod_direction_output(panel->enable_gpio, 0);
+		else
+			gpiod_direction_output(panel->enable_gpio, 1);
 	}
 
 	err = of_drm_get_panel_orientation(dev->of_node, &panel->orientation);
-- 
2.53.0

